dev/PowerNotifications/PowerNotifications.h

```diff
@@ -0,0 +1,158 @@
+    {
+        int   oldBatteryChargePercent, batteryChargePercent;
+        winrt::Microsoft::ProjectReunion::BatteryStatus     oldBatteryStatus, batteryStatus;
+        winrt::Microsoft::ProjectReunion::PowerSupplyStatus oldPowerSupplyStatus, powerSupplyStatus;
+    };
+
+    struct PowerManager
+    {
+        PowerManager() = default;
+
+        using eventType = wil::object_without_destructor_on_shutdown<winrt::event<PowerEventHandle>>;
+
+        static winrt::Microsoft::ProjectReunion::EnergySaverStatus EnergySaverStatus();
+        static EventToken EnergySaverStatusChanged(PowerEventHandle const&);
+        static void EnergySaverStatusChanged(EventToken const&);
+        static void EnergySaverStatusChanged_Callback(::EnergySaverStatus);
+
+        static winrt::Microsoft::ProjectReunion::BatteryStatus BatteryStatus();
+        static EventToken BatteryStatusChanged(PowerEventHandle const&);
+        static void BatteryStatusChanged(EventToken const&);
+        static void CompositeBatteryStatusChanged_Callback(CompositeBatteryStatus*);
+
+        static winrt::Microsoft::ProjectReunion::PowerSupplyStatus PowerSupplyStatus();
+        static EventToken PowerSupplyStatusChanged(PowerEventHandle const&);
+        static void PowerSupplyStatusChanged(EventToken const&);
+
+        static int RemainingChargePercent();
+        static EventToken RemainingChargePercentChanged(PowerEventHandle const&);
+        static void RemainingChargePercentChanged(EventToken const&);
+        
+        static Windows::Foundation::TimeSpan RemainingDischargeTime();
+        static EventToken RemainingDischargeTimeChanged(PowerEventHandle const&);
+        static void RemainingDischargeTimeChanged(EventToken const&);
+        static void RemainingDischargeTimeChanged_Callback(ULONGLONG);
+
+        static winrt::Microsoft::ProjectReunion::PowerSourceStatus PowerSourceStatus();
+        static EventToken PowerSourceStatusChanged(PowerEventHandle const&);
+        static void PowerSourceStatusChanged(EventToken const&);
+        static void PowerSourceStatusChanged_Callback(DWORD);
+
+        static winrt::Microsoft::ProjectReunion::DisplayStatus DisplayStatus();
+        static EventToken DisplayStatusChanged(PowerEventHandle const&);
+        static void DisplayStatusChanged(EventToken const&);
+        static void DisplayStatusChanged_Callback(DWORD);
+
+        static winrt::Microsoft::ProjectReunion::SystemIdleStatus SystemIdleStatus();
+        static EventToken SystemIdleStatusChanged(PowerEventHandle const&);
+        static void SystemIdleStatusChanged(EventToken const&);
+        static void SystemIdleStatusChanged_Callback();
+
+        static winrt::Microsoft::ProjectReunion::PowerSchemePersonality PowerSchemePersonality();
+        static EventToken PowerSchemePersonalityChanged(PowerEventHandle const&);
+        static void PowerSchemePersonalityChanged(EventToken const&);
+        static void PowerSchemePersonalityChanged_Callback(GUID);
```

Comment:
Do we know if there are any events here to differentiate state changes from Disconnected Standby and Connected Standby? That would be useful for scenarios like socket rundown for persistent connection scenarios. Today we kind of struggle with that and completely rely on NLM to tell us that we were disconnected when going into low power.

Reply:
@brialmsft Are we supporting these events?

Reply:
I don't believe any of the events we're planning to implement currently cover that, but we're not experts in the power space. If you think there's good reason to include those events, it may be possible to include them in future versions.

Reply:
I know BASE was trying to solve a very similar problem for one of their APIs recently. CS is technically simply LowPower and the platform exposes APIs for it. Infact the BatterySaver event fires when the device goes into this state. DS was a little weird: We have to use a Network Disconnected event to figure this one out in addition to the lower power state changes. It would be awesome to combine the 2 and have some predictable way in the future to inform app developers.

Reply:
We can aim this for a future release as suggested.